provider/lxd/credentials.go

```diff
@@ -278,9 +294,13 @@ func (p environProviderCredentials) FinalizeCredential(
 	switch authType := args.Credential.AuthType(); authType {
 	case cloud.InteractiveAuthType:
 		credAttrs := args.Credential.Attributes()
-		// We don't care if the password is empty, just that it exists. Empty
-		// passwords can be valid ones...
-		if _, ok := credAttrs[credAttrTrustPassword]; ok {
+		_, isTrustToken := credAttrs[credAttrTrustToken]
+		// We don't care if the password/token is empty, just that it
+		// exists. Empty passwords can be valid ones...
+		// Both trust password and trust tokens should create new client
+		// certs that will be passed to the server and store them.
+		_, isTrustPassword := credAttrs[credAttrTrustPassword]
```

Comment:
we are missing a check that we don't end up with both trust token and trust password
we should probably fail if they supply both due to the ambiguity